src/batcher.rs

```diff
@@ -0,0 +1,148 @@
+use std::{collections::VecDeque, sync::Arc, time::Duration};
+
+use bdk_wallet::bitcoin::{self, Amount};
+use chrono::Utc;
+use kanal::{unbounded_async, AsyncSender, SendError};
+use parking_lot::{RwLock, RwLockWriteGuard};
+use terrors::OneOf;
+use tokio::{
+    select, spawn,
+    task::{spawn_blocking, JoinHandle},
+    time::interval,
+};
+use tracing::{error, info, info_span, Instrument};
+
+use crate::l1::{fee_rate, L1Wallet, Persister, ESPLORA_CLIENT};
+
+pub enum PayoutRequest {
+    L1(L1PayoutRequest),
+}
+
+pub struct L1PayoutRequest {
+    pub address: bitcoin::Address,
+    pub amount: Amount,
+}
+
+pub struct Batcher {
+    task: Option<JoinHandle<()>>,
+    payout_sender: Option<AsyncSender<PayoutRequest>>,
+    cfg: BatcherConfig,
+}
+
+#[derive(Debug)]
+pub struct BatcherNotStarted;
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct BatcherNotAvailable(SendError);
+
+#[derive(Debug, Clone)]
+pub struct BatcherConfig {
+    pub period: Duration,
+    pub max_per_tx: usize,
+    pub max_in_flight: usize,
+}
+
+impl Batcher {
+    pub fn new(cfg: BatcherConfig) -> Self {
+        Self {
+            task: None,
+            payout_sender: None,
+            cfg,
+        }
+    }
```

Comment:
I still don't like this struct being able to be instantiated without a `task`. But it's fine for now.

Reply:
Ok, you've convinced me

Reply:
Might be worthy some docstrings